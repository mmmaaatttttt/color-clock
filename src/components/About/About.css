.About {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  min-height: 100vh;
  flex-direction: column;
  margin: 2rem;
}

.About h2 {
  margin: 0 auto 2rem;
}

.About p,
.About li {
  font-size: 1.75rem;
  letter-spacing: 0.04em;
  margin-bottom: 1.2rem;
}

@media (min-width: 80rem) {
  .About {
    width: 50%;
    margin: 0 auto;
  }
}
