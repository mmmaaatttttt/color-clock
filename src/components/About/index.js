import React from "react";
import { Link } from "react-router-dom";
import "./About.css";

function About() {
  return (
    <div className="About">
      <h2>Make time visible.</h2>
      <p>
        For folks who can't tell time and have a hard time with transitions
        (e.g. small children), color clocks give a clear visual that tells them
        how close they are to an upcoming transition. Time-telling skills not
        required.
      </p>
      <p>
        Color clocks smoothly transition between three different phases: start,
        warning and end. You control what time each phase should start and what
        each phase should look like.
      </p>
      <p>How to make a color clock:</p>
      <ol>
        <li>
          Click <Link to="/">here</Link> to fill out the color clock form.
        </li>
        <li>Select a time for the clock to start.</li>
        <li>
          Select a color for the start time (defaults to{" "}
          <span className="green">green</span>).
        </li>
        <li>
          Repeat steps 2 and 3 for the warning time and the ending time (colors
          default to <span className="yellow">yellow</span> and{" "}
          <span className="red">red</span>, respectively.
        </li>
        <li>Click on the button to create your clock.</li>
        <li>
          (Optional.) Bookmark the clock page to revisit it any day in the
          future. The clock will remember the times and colors you selected!
        </li>
      </ol>
      <p>
        Please note that times must all be for the same calendar day. For
        example, it's not possible to make a clock that starts at 11:30 PM and
        ends at 1:00 AM the next day.
      </p>
      <p>
        <Link to="/">Go back</Link>
      </p>
    </div>
  );
}

export default About;
