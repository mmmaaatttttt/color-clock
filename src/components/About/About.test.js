import React from "react";
import { render } from "@testing-library/react";
import About from "./";

describe("basic About tests", function () {
  it("renders without crashing", function () {
    render(<About />);
  });

  it("matches snapshot", function () {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
