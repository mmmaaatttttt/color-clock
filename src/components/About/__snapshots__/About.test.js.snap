// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About basic tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="About"
  >
    <h2>
      Make time visible.
    </h2>
    <p>
      For folks who can't tell time and have a hard time with transitions (e.g. small children), color clocks give a clear visual that tells them how close they are to an upcoming transition. Time-telling skills not required.
    </p>
    <p>
      Color clocks smoothly transition between three different phases: start, warning and end. You control what time each phase should start and what each phase should look like.
    </p>
    <p>
      How to make a color clock:
    </p>
    <ol>
      <li>
        Click 
        <a
          href="/"
        >
          here
        </a>
         to fill out the color clock form.
      </li>
      <li>
        Select a time for the clock to start.
      </li>
      <li>
        Select a color for the start time (defaults to 
        <span
          style="background-color: rgb(122, 231, 185);"
        >
          green
        </span>
        ).
      </li>
      <li>
        Repeat steps 2 and 3 for the warning time and the ending time (colors default to 
        <span
          style="background-color: rgb(255, 236, 158);"
        >
          yellow
        </span>
         and 
        <span
          style="background-color: rgb(254, 173, 205);"
        >
          red
        </span>
        , respectively).
      </li>
      <li>
        Click on the button to create your clock.
      </li>
      <li>
        (Optional.) Bookmark the clock page to revisit it any day in the future. The clock will remember the times and colors you selected!
      </li>
    </ol>
    <p>
      Please note that times must all be for the same calendar day. For example, it's not possible to make a clock that starts at 11:30 PM and ends at 1:00 AM the next day.
    </p>
    <p>
      <a
        href="/"
      >
        Go back
      </a>
    </p>
  </div>
</DocumentFragment>
`;
